src/inmanta/agent/forking_executor.py

```diff
@@ -159,7 +169,37 @@ def connection_lost(self, exc: Exception | None) -> None:
 
 
 class ExecutorClient(FinalizingIPCClient[ExecutorContext], LogReceiver):
-    pass
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        # Keeps track of when this client was active last
+        self.last_used_at: datetime.datetime = datetime.datetime.now().astimezone()
+
+    def get_idle_time(self) -> timedelta:
+        return datetime.datetime.now().astimezone() - self.last_used_at
+
+    @typing.overload
+    def call(
+        self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: typing.Literal[True] = True
+    ) -> Future[ReturnType]: ...
+
+    @typing.overload
+    def call(self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: typing.Literal[False]) -> None: ...
+
+    def call(self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: bool = True) -> Future[ReturnType] | None:
+        """Call a method with given arguments"""
+        self.last_used_at = datetime.datetime.now().astimezone()
+        response = super().call(method, has_reply)
+        assert response is None or isinstance(response, Future)
+        return response
```

Comment:
This part I'm unsure of as well and I don't understand the resulting mypy error
```
+ src/inmanta/agent/forking_executor.py:192: error: No overload variant of "call" of "IPCClient" matches argument types "IPCMethod[ExecutorContext, ReturnType]", "bool"  [call-overload]
+ src/inmanta/agent/forking_executor.py:192: note: Possible overload variants:
+ src/inmanta/agent/forking_executor.py:192: note:     def [ReturnType] call(self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: Literal[True] = ...) -> Future[ReturnType]
+ src/inmanta/agent/forking_executor.py:192: note:     def [ReturnType] call(self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: Literal[False]) -> None

```

Reply:
https://github.com/python/mypy/issues/14764